models/playlist.ts

```diff
@@ -17,4 +17,41 @@ module.exports.savePlaylist = function (name, songs, user_id, type, startdatetim
             return callback(err, result);
         }
     })
+};
+module.exports.saveOrUpdatePlaylist = function (name, songs, user_id, type, startdatetime, enddatetime, callback) {
+    var playlistFound = false;
+    pool.query("SELECT user_id ,name FROM playlists", function (err, result, fields) {
+        if (err) {
+            Logger.log('error', err);
+            callback(err, '');
+        }
+        else {
+            for (var i = 0; i < result.rows.length; i++) {
```

Comment:
En loop je daar doorheen met NodeJS. Dit is ontzettend in efficient. Je stuurt mogelijk duizenden playlists vanuit de database servers, maakt hier objecten voor aan lokaal en gaat ze dan loopen ... Je kan ook van te voren checken of ie al bestaat. Gewoon select count(*)  as count from playlists WHERE name = AND user_id =